### ISSUE TITLE
eth/tracers: callTracer output traces append with events

### ISSUE THREAD

**Issue by jsvisa** [ethereum/go-ethereum#25967]
# Rationale

Why should this feature exist?

Currently, the `callTracer` only captures the `call, create` traces. And in some cases, we want to capture what is triggered by each trace, eg: call `USDT.tranfer` will trigger out `Transfer(from,  to, value)` event.

What are the use-cases?

As EventLog is often used to record an event that occurred, we can combine the call trace(input) + event(output) to analysis some complex transactions.  eg:

![image](https://user-images.githubusercontent.com/3627395/195141185-d56d17fc-8d5f-41ed-bba9-fb9721f84b27.png)

From https://tx.eth.samczsun.com/ethereum/0x4b119a4f4ba1ad483e9851973719f310527b43f3fcc827b6d52db9f4c1ddb6a2

# Implementation

Do you have ideas regarding the implementation of this feature?

I want to add a new option named `includingLog bool` in callTracerConfig https://github.com/ethereum/go-ethereum/blob/1c737e8b6da2b14111f8224ef3f385b1fe0cd8b9/eth/tracers/native/call.go#L76-L78

Request as below:

```json
{
    "jsonrpc": "2.0",
    "method": "debug_traceTransaction",
    "params": [
        "0xfoobar",
        {
            "tracer": "callTracer",
            "timeout": "120s", 
            "tracerConfig": {"includingLog": true}
        }
    ],
    "id": 67
}
```

And the output looks like this:

```json
{
  "jsonrpc": "2.0",
  "id": 67,
  "result": {
    "from": "0x68e8198d5b3b3639372358542b92eb997c5c314a",
    "gas": "0x1b9e3",
    "gasUsed": "0x126dd",
    "to": "0xaaa4940aa878d932d3482bf1de332e1d50c15aaa",
    "input": "0xc6bbb408000000000000000000000000f35875a064cdbc29d7174f5c699f1ebeaa407036000000000000000000000000000000000000000000182c0261e603cc59ba68a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000041c6a6f84c29a3c0374aeabb39c1ffb3096731ac9605105dad2c98bcc130a1c7bf1b8e20fbc59ec30d60c6261fa1969f553300b0fc81e51ad50b2b0740d4ae56d70000000000000000000000000000000000000000000000000000000000000000",
    "output": "0x0000000000000000000000000000000000000000000000000000000000000001",
    "calls": [
      {
        "from": "0xaaa4940aa878d932d3482bf1de332e1d50c15aaa",
        "gas": "0x1505e",
        "gasUsed": "0xbb8",
        "to": "0x0000000000000000000000000000000000000001",
        "input": "0x5943c71e52b15e98c5147ea35a5d4cbac106daccd712b01403a5899cf95d39e0000000000000000000000000000000000000000000000000000000000000001bc6a6f84c29a3c0374aeabb39c1ffb3096731ac9605105dad2c98bcc130a1c7bf1b8e20fbc59ec30d60c6261fa1969f553300b0fc81e51ad50b2b0740d4ae56d7",
        "output": "0x00000000000000000000000068e8198d5b3b3639372358542b92eb997c5c314a",
        "type": "STATICCALL"
      },
      {
        "from": "0xaaa4940aa878d932d3482bf1de332e1d50c15aaa",
        "gas": "0x1097e",
        "gasUsed": "0x787f",
        "to": "0xaaa7a10a8ee237ea61e8ac46c50a8db8bcc1baaa",
        "input": "0xa9059cbb000000000000000000000000f35875a064cdbc29d7174f5c699f1ebeaa40703600000000000000000000000000000000000000000017b03fc6572caea9da8542",
        "output": "0x0000000000000000000000000000000000000000000000000000000000000001",
        "value": "0x0",
        "type": "CALL",
        "logs": [
          {
            "topics": [
              "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
              "0x000000000000000000000000aaa4940aa878d932d3482bf1de332e1d50c15aaa",
              "0x000000000000000000000000f35875a064cdbc29d7174f5c699f1ebeaa407036"
            ],
            "data": "0x0000000000000000000000000000000000000000000000000000000000000001"
          }
        ]
      }
    ],
    "value": "0x0",
    "type": "CALL"
  }
}
```

Are you willing to implement this feature?

Yes, 🙋🏼

**Comment by holiman** [comment]
Sounds reasonable. I am not sure if it's better to add functionality into existing tracers, with custom configurations, or simply adding new tracers. 
But I guess maybe if you implement it, we can see which feels nicest.

**Comment by jsvisa** [comment]
I'll try to implement it ASAP

**Comment by jsvisa** [comment]
implemented in #25991
